### ISSUE TITLE
CarbonInterval does not honor setting "toStringFormat".

### ISSUE THREAD

**Issue by digitaladapt** [briannesbitt/Carbon#2002]
Hello,

I encountered an issue with the following code:
```php
echo CarbonInterval::createFromFormat('H:i:s', '01:02:03')->settings(['toStringFormat' => '%H:%I:%S']);
```

Carbon version: **2.29.1**

PHP version: **7.2.24**

I expected to get:

```
"01:02:03"
```

But I actually get:

```
"1 hour 2 minutes 3 seconds"
```

CarbonInterval never checks settings for formatting override, and instead unconditionally uses the function $this->forHumans(), rather than respecting the value set in $this->localToStringFormat.

Purposed patch attached (use existing Converter Trait):

[CarbonInterval-ToStringFormat.txt](https://github.com/briannesbitt/Carbon/files/4119371/CarbonInterval-ToStringFormat.txt)

Thanks!

**Comment by kylekatarnls** [comment]
I'm OK for this new feature. Would you like to submit a pull-request so your authorship would be saved in [contributors page](https://github.com/briannesbitt/Carbon/graphs/contributors)?

**Comment by kylekatarnls** [comment]
Will be fixed in 2.30.
